include/oneapi/dpl/pstl/execution_defs.h

```diff
@@ -228,6 +228,9 @@ struct __range_traits
 template <typename _R>
 using __value_t = typename __range_traits<_R>::__value_t;
 
+template <typename _Proj, typename _ValueType>
+using __key_t = decltype(::std::declval<_Proj&>()(::std::declval<_ValueType&>()));
```

Comment:
I am not sure if it's right way of determine `__key_t`. I mean why `_ValueType&`? as far as I understand it's not guaranteed that dereferencing range would always give and lvalue